src/main/java/uk/gov/ons/ctp/response/notify/client/NotificationClientFactory.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.ons.ctp.response.notify.client;
+
+import jodd.util.StringUtil;
+import lombok.extern.slf4j.Slf4j;
+import uk.gov.ons.ctp.response.notify.config.NotifyConfiguration;
+import uk.gov.service.notify.NotificationClientApi;
+
+/**
+ * A simple factory class to create a NotificationClientApi object from a NotifyConfiguration
+ */
+@Slf4j
+public class NotificationClientFactory {
+
+    /**
+     * An enum to represent the different possible configurations.  Parsed from notify.debugType application.yml
+     * configuration item (or env etc)
+     * None - default, uses ConfigurationAwareNotificationClient
+     * NotifyClientException - uses an ExceptionThrowingNotificationClient with an optional supplied httpCode
+     * RuntimeException - uses RuntimeExceptionThrowingNotificationClient
+     */
+    public enum DebugType {
+        None, NotifyClientException, RuntimeException;
+    }
+
+    public static NotificationClientApi getNotificationClient(NotifyConfiguration config){
+        NotificationClientApi client = null;
+        DebugType type = DebugType.None;
+        String debugTypeString = config.getDebugType();
+
+        if (StringUtil.isBlank(debugTypeString) == false){
+            type = DebugType.valueOf(debugTypeString);
+        }
+
+        switch(type){
```

Comment:
Why do we need both ExceptionThrowingNotificationClient and RuntimeExceptionThrowingNotificationClient?

Reply:
I couldn't think of a way to do it as 1 class - I did try - the checked exceptions thrown by the NotificationClientApi constrain the options somewhat.  The best I could come up with was a class with both a NotificationClientException and a RuntimeException as members and it would throw whichever was not null.  I don't think this is any more elegant than 2 classes

Reply:
And to actually answer your original question, the code deliberately handles RuntimeExceptions differently to other exceptions so we need both classes to be able to produce both situations.

Reply:
Re. Is there some value in adding some tests around the bug fixes?

Not sure how to go about this - the problem was that Spring Integration got twisted when given a certain set of inputs. To build tests around this we'd have to persuade Spring Integration to use a mocked RabbitMQ then count how many messages were in each queue.  Any ideas?

Reply:
Is it possible to write an integration test and would that catch this issue?